### PR TITLE
Revert "AMDGPU: Don't avoid clamp of bit shift in BFE pattern (#115372)"

### PR DESCRIPTION
**Pull request by changpeng**
Based on the suggestion from https://github.com/llvm/llvm-project/pull/115543, we should not do the pattern matching from x << (32-y) >> (32-y) to "bfe x, 0, y" at all.
This reverts commits a2bacf8ab58af4c1a0247026ea131443d6066602 and https://github.com/llvm/llvm-project/commit/bdf8e308b7ea430f619ca3aa1199a76eb6b4e2d4.

### CODE REVIEW

**Review by jayfoad — COMMENTED** [review]
Please say why in the description.

File: llvm/test/CodeGen/AMDGPU/extract-lowbits.ll [hunk 1]
```diff
@@ -150,39 +150,22 @@ define i32 @bzhi32_c4_commutative(i32 %val, i32 %numlowbits) nounwind {
 ; ---------------------------------------------------------------------------- ;
 
 define i32 @bzhi32_d0(i32 %val, i32 %numlowbits) nounwind {
-; GCN-LABEL: bzhi32_d0:
```
Comment (arsenm) [inline llvm/test/CodeGen/AMDGPU/extract-lowbits.ll:153]: Should keep the test, and also add a variant with canonical IR

Comment (changpeng) [reply llvm/test/CodeGen/AMDGPU/extract-lowbits.ll:153]: This test is kept now, thanks.

I think the canonical IR test is there in the test:
define i32 @bzhi32_d0(i32 %val, i32 %numlowbits) nounwind {
  %numhighbits = sub i32 32, %numlowbits
  %highbitscleared = shl i32 %val, %numhighbits
  %masked = lshr i32 %highbitscleared, %numhighbits
  ret i32 %masked
}

**Comment by arsenm** [comment]
> > Please say why in the description.
> 
> We are going to revert the original patch for bfe patterns. We believe the optimization is just the corner cases and does not deserve the cost of compilation time.

This comment is not part of the commit description

**Comment by changpeng** [comment]
> > > Please say why in the description.
> > 
> > 
> > We are going to revert the original patch for bfe patterns. We believe the optimization is just the corner cases and does not deserve the cost of compilation time.
> 
> This comment is not part of the commit description

I know.  Already updated the commit description (in different words though)

**Pull request merged**